### ISSUE TITLE
Column toggle doesn't work properly

### ISSUE THREAD

**Issue by FrancescoBorzi** [swimlane/ngx-datatable#1194]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

```
<ngx-datatable
        class="table table-striped table-bordered"
        #table
        [rows]="rows"
        [columnMode]="'force'"
        [headerHeight]="50"
        [footerHeight]="50"
        [rowHeight]="'auto'"
        [limit]="limit"
        [selectionType]="'checkbox'"
        [selected]="selectedRows"
        (select)="updateSelectedRows($event)">
        <ngx-datatable-column *ngIf="myBooleanVariable" prop="myProp" name="My Column Name">
        </ngx-datatable-column>
       <!--other columns-->
</ngx-datatable>
```

**Current behavior**

When `myBooleanVariable`'s value is changed, the column will be toggled only after that I mouseover the datatable.

**Expected behavior**

When `myBooleanVariable`'s value is changed, the column should be toggled immediately, no matter if I mousehover the datatable or not.

**Reproduction of the problem**

See the code above.

**What is the motivation / use case for changing the behavior?**

I guess somehow the datatable should detect the change and re-render itself, instead it will only re-render when it's being mousehovered.

**Please tell us about your environment:**

```
Angular CLI: 1.6.0
Node: 8.9.1
OS: linux x64
Angular: 5.1.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.6.0
@angular-devkit/build-optimizer: 0.0.35
@angular-devkit/core: 0.0.22
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0
@schematics/schematics: 0.0.10
typescript: 2.5.3
webpack: 3.10.0

```

* **Table version:** 11.0.4

* **Angular version:** 5.1.0

* **Browser:** Chrome

* **Language:** TypeScript 2.5.3

**Comment by Asfath** [comment]
did u find any solution..? i have the same issue

**Comment by ShinDarth** [comment]
@Asfath not yet :( 

@amcdnl can you please have a look at this bug?

**Comment by tabuckner** [comment]
@FrancescoBorzi @amcdnl I am also running into this bug in a legacy app that consumes `ngx-datatable`.

Any updates?

**Comment by tabuckner** [comment]
As a follow up, you can view part of the bug in the column toggling example at:

http://swimlane.github.io/ngx-datatable/#toggle

You are unable to toggle off all columns in this example.

**Comment by joeveiga** [comment]
Duplicate of #1581
